LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/MessageDispatcher.cs

```diff
@@ -96,9 +97,16 @@ void DispatchLoRaDataMessage(LoRaRequest request)
             this.deviceRegistry.GetLoRaRequestQueue(request).Queue(request);
         }
 
-        bool IsValidNetId(byte devAddrNwkid, uint netId)
+        bool IsValidNetId(LoRaPayloadData loRaPayload)
         {
-            var netIdBytes = BitConverter.GetBytes(netId);
+            var currentDevAddr = ConversionHelper.ByteArrayToString(loRaPayload.DevAddr);
```

Comment:
I would switch the checks here. 99% of the servers won't be using the devAddr whitelist. Therefore I'd check first the correct netId and then the whitelist.

Reply:
Good point thank you

Reply:
Cool I have seen you changed, thanks!

There is a minor optimisation yet, you could remove the allocation in case there is no whitelist defined (which will 99% of the cases):
```c#
if (this.configuration.AllowedDevAddresses != null &&
this.configuration.AllowedDevAddresses.Count > 0 &&
 this.configuration.AllowedDevAddresses.Contains(ConversionHelper.ByteArrayToString(loRaPayload.DevAddr)))
{
}
```

I will approve anyway, you can add to a next PR if you want.